pydynamo/storage/disk/diskstore.py

```diff
@@ -52,86 +59,200 @@ def set_index_ratio(self, new_index_ratio: float) -> None:
 
-    def create_sstable(self) -> SSTable:
+    def flush(self) -> None:
         """
-        :return: create a sstable
+        :return: None
+        if it is not flushing and size of expected sstable
+        file reaches threshold, set start_flush to True to
+        block read and write flush mem_table into disk,
+        add sstable in ss_tables update id, clean mem_table
+        and index_table.
+        after it, set start_flush back to False
         """
-        size = self._flush_to_disk()
-        new_sstable = SSTable(self.store_name, self.id, size,
-                              self.index_table, self.path, self.last_index)
+        self.ss_tables.append(self.create_sstable(self.mem_table, {}))
         self.id += 1
         self.mem_table.clean()
         self.index_table = []
-        return new_sstable
+
+    def create_sstable(self, mem_table: InMemoryStore,
+                       time_stamp: dict) -> SSTable:
+        """
+        :param mem_table: table in memory
+        :param time_stamp: time stamp corresponding to mem_table
+        :return: create a sstable
+        """
+        size = self._flush_mem_to_disk(mem_table, time_stamp)
+        new_ss_table = SSTable(self.store_name, self.id, size,
+                               self.index_table, self.path, self.last_index)
+        return new_ss_table
 
     def set(self, key: str, value: str) -> Any:
-        pass
+        """
+        :param key: key to add
+        :param value: value corresponding to key
+        :return: add the set of data
+        """
+        self.rwlock.wlock_acquire()
+        self.mem_table.set(key, value)
+        if self.mem_table.get_size() >= self.mem_size_threshold:
+            self.flush()
+        self.rwlock.wlock_release()
 
     def get(self, key: str) -> Any:
-        pass
+        """
+        :param key: the key to be found
+        :return: the value corresponding to the key. if not found, return None
+        check whether it starts flushing. if flushing, raise an exception.
+        otherwise, read mem_table, return value if found.
+        if not found in mem_table, read sstables and find
+        the value with latest timestamp
+        """
+        self.rwlock.rlock_acquire()
+        value = None
+        if self.mem_table.contain_key(key):
+            if self.mem_table.is_removed(key):
+                self.rwlock.rlock_release()
+                return None
+            else:
+                value = self.mem_table.get(key)
+        else:
+            timestamp = 0
+            for ss_table in self.ss_tables:
+                if ss_table.contain(key):
+                    cur_timestamp = ss_table.get_timestamp(key)
+                    if cur_timestamp > timestamp:
+                        timestamp = cur_timestamp
+                        value = ss_table.get(key)
+        self.rwlock.rlock_release()
+        return value
 
     def iterator(self) -> Any:
-        pass
+        self.rwlock.wlock_acquire()
```

Comment:
why do we need to flush when we call iterator? shouldn't the iterator be able to iterate through sstables and memtables?